src/_Problems_/balanced-parentheses/index.js

```diff
@@ -0,0 +1,27 @@
+/* 
+    Given a string containing characters and "(" , ")" determine if input string is valid if:
+    -  Open parantheses is closed by same number & type of parantheses.
+    Note that an empty string is also considered valid and any alphanumeric char can be inserted in between pratheses.
+
+    Example : 
+       ((())) is balanced.
+       ((() is not balanced.
+       )( is not balanced.
+    We will solve this problem using ES6 - Array Method reduce().
+ */
+
+function balancedParantheses(string) {
+  return !string.split('').reduce(function(prev, char) {
+    if (prev < 0) return prev;
+
+    if (char === '(') return ++prev;
+
+    if (char === ')') return --prev;
+
+    return prev;
```

Comment:
Can we use a **switch case** here?

Reply:
I think switch case isn't needed here since we have to check two variables char & prev here. can i use ternary operator instead ?

Reply:
Something like this

```js
  return !string.split('').reduce((prev, char) => {
    const result = prev < 0 ? prev : char === '(' ? ++prev : char === ')' ? --prev : prev;
    return result;
  }, 0);
```

Reply:
The above snippet is not very readable.

Reply:
so, if else pattern will be fine right ? switch case will add more code since we have to check cases for prev & char.